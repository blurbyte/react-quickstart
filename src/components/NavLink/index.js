import { Link } from 'react-router';
import styled from 'styled-components';

const NavLink = styled(Link)`
  font-family: 'OpenSans Light', Arial, Helvetica, sans-serif;
  color: #ff4326;
  font-size: 1.4rem;
  text-transform: uppercase;
  display: block;
  margin-right: 1.4rem;

  &:hover {
    text-decoration: underline;
  }
`;

export default NavLink;
