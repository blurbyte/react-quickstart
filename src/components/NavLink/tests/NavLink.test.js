import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { MemoryRouter } from 'react-router-dom';

import NavLink from '../index';

describe('<NavLink />', () => {
  it('should render correctly', () => {
    // props init
    const props = {
      to: '/counter'
    };
    // component init
    const component = renderer.create(
      <MemoryRouter>
        <NavLink {...props} />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
