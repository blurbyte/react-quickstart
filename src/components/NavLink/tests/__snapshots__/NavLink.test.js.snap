// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> should render correctly 1`] = `
.gmsRav {
  font-family: 'OpenSans Light', Arial, Helvetica, sans-serif;
  color: #ff4326;
  font-size: 1.4rem;
  text-transform: uppercase;
  display: block;
  margin-right: 1.4rem;
}

.gmsRav:hover {
  text-decoration: underline;
}

<a
  className="gmsRav"
  href="/counter"
  onClick={[Function]}
/>
`;
